src/main/haskell/kore/src/Kore/Step/Simplification/Data.hs

```diff
@@ -109,30 +200,112 @@ instance HasLog Environment LogMessage (ExceptT e Simplifier) where
                 res <- runExceptT (action msg)
                 return $ either (const ()) id res
 
-{- | Run a simplifier computation.
+{- | Evaluate a simplification, returning results along all branches.
+ -}
+evalSimplifierBranch
+    :: LogAction Simplifier LogMessage
+    -> Simplifier a
+    -> SMT [a]
+evalSimplifierBranch logger (Simplifier simpl) =
+    withSolver' $ \solver -> do
+        let env = Environment solver logger
+        runReaderT (toReverseList simpl) env
+
+{- | Run a simplification, returning the result of only one branch.
```

Comment:
Why would this be useful? From its name, this seems a rather innocent function, but it crashes in an unexpected way, so I'm curious what we would gain.

Reply:
This is useful temporarily to add internal branching without changing all the tests prematurely. (Right now this is just one commit adding the capability to branch internally. Nothing uses internal branches yet, so this is safe to use in all the tests.) By the time the pull request is ready, I imagine it will go away.